activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -55,17 +55,15 @@ def initialize(files, dirs = {}, &block)
       dtw = directories_to_watch
       @dtw, @missing = dtw.partition(&:exist?)
 
-      if @dtw.any?
-        # Loading listen triggers warnings. These are originated by a legit
-        # usage of attr_* macros for private attributes, but adds a lot of noise
-        # to our test suite. Thus, we lazy load it and disable warnings locally.
-        silence_warnings do
-          require "listen"
-        rescue LoadError => e
-          raise LoadError, "Could not load the 'listen' gem. Add `gem 'listen'` to the development group of your Gemfile", e.backtrace
-        end
+      # Loading listen triggers warnings. These are originated by a legit
+      # usage of attr_* macros for private attributes, but adds a lot of noise
```

Comment:
Looks like `private attr_reader` aren't warnings since Ruby 2.3: https://github.com/ruby/ruby/commit/32a5a098e39e377350e4696d2d9f2d8930f1ec41

So maybe we can ditch this `silence_warnings` too.

Reply:
Running `bundle exec rake test TEST=test/evented_file_update_checker_test.rb` with `silence_warnings` removed doesn't seem to spit out any warnings. 👍 to removing.

Reply:
Good to know!, I removed loaded the `listen` gem at the top of the file as asked by @rafaelfranca